### PR TITLE
Improve error handling for #282

### PR DESCRIPTION
**Pull request by juliantoledo**
Reject manifest if it does not pass the Lighthouse manifest-parser validation and show the error from it.

I also updated the manifest-parser.

### CODE REVIEW

**Comment by juliantoledo** [comment]
PTAL

File: lib/pwa.js [hunk 1]
```diff
@@ -336,8 +336,10 @@ exports._save = function(pwa) {
 
       // Validates the manifest or, reject and return errors in an array
       const res = libManifest.validateManifest(manifest.raw, pwa.manifestUrl, pwa.absoluteStartUrl);
-      pwa.manifest = res.length === 0 ? manifest : Promise.reject(res);
-
+      if (res.length !== 0) {
+        return Promise.reject(res);
```
Comment (juliantoledo) [inline lib/pwa.js]: res contains the rejection information from the validator, I could add an additional message to it.

File: third_party/manifest-parser.js [hunk 2]
```diff
@@ -157,7 +157,7 @@ function parseStartUrl(jsonInput, manifestUrl, documentUrl) {
 }
 
 function parseDisplay(jsonInput) {
-  let display = parseString(jsonInput.display, true);
+  const display = parseString(jsonInput.display, true);
```
Comment (ithinkihaveacat) [inline third_party/manifest-parser.js]: @juliantoledo Did you use https://github.com/GoogleChrome/gulliver/blob/master/third_party/install.sh to generate this? Just wondering why the updated `Color.js` isn't here as well. (Which seems to include some awkward changes, but still probably better to commit both together.)

Comment (juliantoledo) [inline third_party/manifest-parser.js]: @ithinkihaveacat I also updated our script to work with their latest version of both files.

**Pull request merged**